src/dev-server.js

```diff
@@ -0,0 +1,89 @@
+import 'babel-polyfill';
+import {DockerExecServer,DockerExecClient} from 'docker-exec-websocket-server';
+import Docker from 'dockerode-promise';
+import express from 'express';
+import http from 'http';
+import fs from 'fs';
+import slugid from 'slugid';
+import path from 'path';
+import reactViews from 'express-react-views'
+
+var debug = require('debug')('dev-server:');
+
+const PORT = 34678;
+const DOCKER_SOCKET = '/var/run/docker.sock'
+const PATH = '/'+slugid.v4();
+
+var dockerServer, container, docker,server,dockerClient;
+//check if docker is running i.e socket is available
+var start = async ()=>{
+  debug('Process started');
+  if(!fs.statSync(DOCKER_SOCKET).isSocket()){
+    debug("Docker is not running");
+    throw new Error("Docker is not running");
+  }
+  //Set up the docker server
+  docker = new Docker({socketPath:DOCKER_SOCKET});
+  await docker.pull('ubuntu');
+  container = await docker.createContainer({
+    Image:'gdb-test',
+    cmd: ['sleep','600']
+  });
+
+  await container.start();
+  debug("Container started");
+  server = http.createServer();
+  server.listen(PORT);
+  debug("Server listening...")
+  dockerServer = new DockerExecServer({
+    server: server,
+    containerId: container.id,
+    path: PATH
+  });
+  debug("Docker server created");
+
+  //set up express
+  var app = express();
+  app.set('views',__dirname+'client');
+  app.set('view engine','jsx');
+  app.engine('jsx',reactViews.createEngine());
+  http.createServer(app).listen(3000,()=>{
+    debug("http server started");
+  });
+  app.get('/',(req,res)=>{
+    res.render('index',{url:"ws://localhost:"+PORT+PATH});
+  });
+
+  return new Promise(function(resolve, reject) {
+    dockerServer.on('exit',()=>{
+      return resolve();
```

Comment:
Trick 1: You can do: `new Promise((resolve, reject) => {..` (use arrow functions...
Trick 2: You can do: `dockerServer.on('exit', reject);` and `dockerServer.on('error', reject);`
Trick 3: You can do: `dockerServer.once('exit', reject);` and `dockerServer.once('error', reject);`

Number 3 doesn't actually do much :) because calling reject multiple times does nothing to change the promise.